modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ImportAndDeleteObjectEntryPerformanceTest.java

```diff
@@ -0,0 +1,219 @@
+import com.liferay.object.service.ObjectDefinitionLocalService;
+import com.liferay.object.service.ObjectEntryLocalService;
+import com.liferay.petra.string.StringUtil;
+import com.liferay.portal.kernel.json.JSONArray;
+import com.liferay.portal.kernel.json.JSONFactory;
+import com.liferay.portal.kernel.json.JSONFactoryUtil;
+import com.liferay.portal.kernel.json.JSONObject;
+import com.liferay.portal.kernel.json.JSONUtil;
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.test.rule.AssumeTestRule;
+import com.liferay.portal.kernel.test.rule.DataGuard;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.kernel.util.LoggingTimer;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.util.PropsValues;
+
+import java.util.List;
+
+import org.junit.Assume;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Lily Chi
+ */
+@DataGuard(scope = DataGuard.Scope.NONE)
+@RunWith(Arquillian.class)
+public class ImportAndDeleteObjectEntryPerformanceTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new AssumeTestRule("assume"), new LiferayIntegrationTestRule());
+
+	public static void assume() {
+		Assume.assumeTrue(Validator.isNull(System.getenv("JENKINS_HOME")));
+	}
+
+	@BeforeClass
+	public static void setUpClass() throws Exception {
+		_objectFolderJSON = StringUtil.read(
+			ImportAndDeleteObjectEntryPerformanceTest.class.getClassLoader(),
+			"/object-folder-definition.json");
+	}
+
+	@Before
+	public void setUp() throws Exception {
+		_importObjectFolder();
+	}
+
+	@Test
+	public void testDeleteObjectEntry() throws Exception {
+		_importObjectEntry();
+		_deleteObjectEntry();
+	}
+
+	private String _createObjectEntryJSON() throws Exception {
+		if (_OBJECT_ENTRY_COUNT < 1) {
+			throw new IllegalArgumentException(
+				"Must at least create 1 Object Entry!");
+		}
+
+		JSONArray jsonArray = JSONFactoryUtil.createJSONArray();
+
+		for (int i = 0; i < _OBJECT_ENTRY_COUNT; i++) {
+			jsonArray.put(JSONUtil.put("alpha", "foo"));
+		}
+
+		return jsonArray.toString();
+	}
+
+	private void _deleteObjectEntry() {
+		try (LoggingTimer loggingTimer = new LoggingTimer(
+				"Delete " + _OBJECT_ENTRY_COUNT + " Object Entries")) {
+
+			JSONArray jsonArray = _getObjectEntryIdJSONArray();
+
+			HttpInvoker httpInvoker = HttpInvoker.newHttpInvoker();
+
+			httpInvoker.body(jsonArray.toString(), "application/json");
+			httpInvoker.userNameAndPassword(_USER_NAME_AND_PASSWORD);
+			httpInvoker.httpMethod(HttpInvoker.HttpMethod.DELETE);
+			httpInvoker.path(_PATH);
+
+			httpInvoker.invoke();
+
+			while (_currentObjectEntryCount != 0) {
```

Comment:
@slnn Please make `_currentObjectEntryCount` local variables, not a field. For example, when you delete, you know how many entries needs to be deleted, so you shouldn't rely on the count resulted from the import counting.